pyccel/parser/syntax/openmp.py

```diff
@@ -47,81 +49,178 @@ def expr(self):
             return stmt.expr
         elif isinstance(stmt, OmpSingleConstruct):
             return stmt.expr
+        elif isinstance(stmt, OmpCriticalConstruct):
+            return stmt.expr
+        elif isinstance(stmt, OmpBarrierConstruct):
+            return stmt.expr
+        elif isinstance(stmt, OmpMasterConstruct):
+            return stmt.expr
+        elif isinstance(stmt, OmpMaskedConstruct):
+            return stmt.expr
+        elif isinstance(stmt, OmpTaskLoopConstruct):
+            return stmt.expr
+        elif isinstance(stmt, OmpSimdConstruct):
+            return stmt.expr
+        elif isinstance(stmt, OmpAtomicConstruct):
+            return stmt.expr
+        elif isinstance(stmt, OmpTaskWaitConstruct):
+            return stmt.expr
+        elif isinstance(stmt, OmpTaskyieldConstruct):
+            return stmt.expr
+        elif isinstance(stmt, OmpTaskConstruct):
+            return stmt.expr
+        elif isinstance(stmt, OmpFlushConstruct):
+            return stmt.expr
+        elif isinstance(stmt, OmpCancelConstruct):
+            return stmt.expr
+        elif isinstance(stmt, OmpTargetConstruct):
+            return stmt.expr
+        elif isinstance(stmt, OmpTeamsConstruct):
+            return stmt.expr
+        elif isinstance(stmt, OmpDistributeConstruct):
+            return stmt.expr
+        elif isinstance(stmt, OmpSectionConstruct):
+            return stmt.expr
+        elif isinstance(stmt, OmpSectionsConstruct):
+            return stmt.expr
         else:
             raise TypeError('Wrong stmt for OpenmpStmt')
 
 class OmpParallelConstruct(BasicStmt):
-    """Class representing a ."""
+    """Class representing a parallel construct."""
     def __init__(self, **kwargs):
         """
         """
-        self.clauses = kwargs.pop('clauses')
+        self.clauses  = kwargs.pop('clauses')
+        self.combined = kwargs.pop('combined')
 
-        super(OmpParallelConstruct, self).__init__(**kwargs)
+        super().__init__(**kwargs)
 
     @property
     def expr(self):
```

Comment:
If this property always returns the same value, it would be better to do the calculations once, either in the `__init__` or in a function called by `__init__`